bindings/bindingtester/tests/directory_util.py

```diff
@@ -27,89 +27,34 @@
-    dir_id = 0  # Used for debugging
-
-    def __init__(self, is_directory, is_subspace, has_known_prefix=True, path=(), root=None):
-        self.root = root or self
-        self.path = path
-        self.is_directory = is_directory
-        self.is_subspace = is_subspace
-        self.has_known_prefix = has_known_prefix
-        self.children = {}
-
-        self.dir_id = DirListEntry.dir_id + 1
-        DirListEntry.dir_id += 1
-
-    def __repr__(self):
-        return 'DirEntry %d %r: %d' % (self.dir_id, self.path, self.has_known_prefix)
-
-    def add_child(self, subpath, default_path, root, child):
-        if default_path in root.children:
-            # print 'Adding child %r to default directory %r at %r' % (child, root.children[DirectoryTest.DEFAULT_DIRECTORY_PATH].path, subpath)
-            c = root.children[default_path]._add_child_impl(subpath, child)
-            child.has_known_prefix = c.has_known_prefix and child.has_known_prefix
-            # print 'Added %r' % c
-
-        # print 'Adding child %r to directory %r at %r' % (child, self.path, subpath)
-        c = self._add_child_impl(subpath, child)
-        # print 'Added %r' % c
-        return c
-
-    def _add_child_impl(self, subpath, child):
-        # print '%d, %d. Adding child (recursive): %s %s' % (self.dir_id, child.dir_id, repr(self.path), repr(subpath))
-        if len(subpath) == 0:
-            self.has_known_prefix = self.has_known_prefix and child.has_known_prefix
-            # print '%d, %d. Setting child: %d' % (self.dir_id, child.dir_id, self.has_known_prefix)
-            self._merge_children(child)
-
-            return self
-        else:
-            if not subpath[0] in self.children:
-                # print '%d, %d. Path %s was absent (%s)' % (self.dir_id, child.dir_id, repr(self.path + subpath[0:1]), repr(self.children))
-                subdir = DirListEntry(True, True, path=self.path + subpath[0:1], root=self.root)
-                subdir.has_known_prefix = len(subpath) == 1
-                self.children[subpath[0]] = subdir
-            else:
-                subdir = self.children[subpath[0]]
-                subdir.has_known_prefix = False
-                # print '%d, %d. Path was present' % (self.dir_id, child.dir_id)
-
-            return subdir._add_child_impl(subpath[1:], child)
-
-    def _merge_children(self, other):
-        for c in other.children:
-            if c not in self.children:
-                self.children[c] = other.children[c]
-            else:
-                self.children[c].has_known_prefix = self.children[c].has_known_prefix and other.children[c].has_known_prefix
-                self.children[c]._merge_children(other.children[c])
-
-
 def setup_directories(instructions, default_path, random):
-    dir_list = [DirListEntry(True, False, True)]
+    # Clients start with the default directory layer in the directory list
+    DirectoryStateTreeNode.reset()
+    dir_list = [DirectoryStateTreeNode.get_layer('\xfe')]
```

Comment:
I think an argument could be made that these tests should be Python 3 compatible, but I think this use follows the already existing conventions of this file. We can raise a separate issue to deal with compatibility if we want.

Reply:
Sure, making all of these python 3 compatible sounds like a pain. I still might prefer the extra `b` here (and other places) to be more explicit that these are byte strings rather than text strings.

Reply:
I can agree with that, but I think it's out of scope of this change.